python/mspasspy/db/database.py

```diff
@@ -3804,8 +4421,112 @@ def index_mseed_file(self, dfile, dir=None, collection='wf_miniseed'):
+        would be discarded.
+        The dataframe will first call dropna to eliminate None values stored
+        in it, so that after it is transfered into a document, the memory
+        usage will be reduced.
+        :param collection:  MongoDB collection name to be used to save the
+        (often subsetted) tuples of filename as documents in this collection.
+        :param one_to_one: a boolean to control if the set should be filtered by
+        rows.  The default is True which means every row in the dataframe will
+        create a single MongoDB document. If False the (normally reduced) set
+        of attributes defined by attributes_to_use will be filtered with the
+        panda/dask dataframe drop_duplicates method before converting the
+        dataframe to documents and saving them to MongoDB.  That approach
+        is important, for example, to filter things like Antelope "site" or
+        "sitechan" attributes created by a join to something like wfdisc and
+        saved as a text file to be processed by this function.
+        :df: Pandas.Dataframe object, the input to be transfered into mongodb
+        documents
+        :parallel:  a boolean that determine if dask api will be used for operation
+        on the dataframe
+        :param null_values:  is an optional dict defining null field values.
+        When used an == test is applied to each attribute with a key
+        defined in the null_vlaues python dict.  If == returns True, the
+        value will be set as None in dataframe. If your table has a lot of null
+        fields this option can save space, but readers must not require the null
+        field.  The default is None which it taken to mean there are no null
+        fields defined.
+        :param one_to_one: a boolean to control if the duplicate should be deleted
+        :return:  integer count of number of documents added to collection
+        """
+        dbcol = self[collection]
+
+        if parallel:
+            df = daskdf.from_pandas(df, chunksize=1, sort=False)
+
+        if not one_to_one:
+            df = df.drop_duplicates()
+            
+        #   For those null values, set them to None
+        df = df.astype(object)
+        if null_values is not None:
+            for key, val in null_values.items():
+                if key not in df:
+                    continue
+                else:
+                    df[key] = df[key].mask(df[key] == val, None)
+
+        '''
+        if parallel:
+            df = daskdf.from_pandas(df, chunksize=1, sort=False)
+            df = df.apply(lambda x: x.dropna(), axis=1).compute()
+        else:
+            df = df.apply(pd.Series.dropna, axis=1)
+        '''
+        if parallel:
+            df = df.compute()
+            
+        df = df.apply(pd.Series.dropna, axis=1)
+        doc_list = df.to_dict(orient="records")
+        if len(doc_list):
+            dbcol.insert_many(doc_list)
+        return len(doc_list)
+
+    def save_textfile(
```

Comment:
Although it appears redundant, we do want to add all the parameter definitions in the docstrings here.